app/Http/Middleware/AuthenticateWithApiKey.php

```diff
@@ -7,22 +7,36 @@
 
 class AuthenticateWithApiKey
 {
+    protected $keys;
+
+    public function __construct()
+    {
+        $this->keys = [
+      'X-DS-Importer-API-Key' => config('app.callpower_key'),
+      'X-DS-CallPower-API-Key' => config('app.callpower_key'),
+      'X-DS-SoftEdge-API-Key' => config('app.softedge_key'),
+    ];
+    }
+
     /**
      * Handle an incoming request.
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
+     * @param  $headers
      * @return mixed
      *
      * @throws \Illuminate\Auth\AuthenticationException
      */
-    public function handle($request, Closure $next)
+    public function handle($request, Closure $next, ...$headers)
     {
-        // Check the `X-DS-Importer-API-Key` header for authorization.
-        if ($request->header('X-DS-Importer-API-Key') !== config('app.api_key')) {
-            throw new AuthenticationException;
+        // Check the header for authorization.
+        foreach ($headers as $header) {
+            if ($request->header($header) === $this->keys[$header]) {
+                return $next($request);
+            }
         }
 
-        return $next($request);
+        throw new AuthenticationException;
     }
```

Comment:
the indentation is SO weird in this file but without this, the Style CI tests wouldn't pass! I don't know what is wrong with this??

Reply:
Ok I think this looks better and is passing!